spec/function.dd

```diff
@@ -1242,9 +1242,18 @@ int foo(in int x, out int y, ref int z, int q);
     for its type)
 
     $(TROW $(D ref),   parameter is passed by reference)
-    $(TROW $(D scope), references in the parameter
-    cannot be escaped (e.g. assigned to a global variable).
-    Ignored for parameters with no references)
+    $(TROW $(D scope), $(ARGS
+    The parameter does not escape the function call
+    (e.g. by being assigned to a global variable).
+    Ignored for any parameter that is not of reference type.
+    Escape analysis is done for `@safe` functions, for other functions `scope` semantics must
+    be manually enforced.
+    As the parameter does not escape, the compiler can avoid heap-allocation of a unique argument
+    to a `scope` parameter. Due to this, such a function call can become `@nogc`.
+    This is typically done when passing a delegate literal to a `scope` parameter.
+    $(B Note): Escape analysis is currently only done by `dmd`
+    when the `-dip1000` switch is passed.
```

Comment:
This statement isn't exactly true, there is some scope analysis without `-dip1000` (esp. given that [`-dip25` is now partially enabled by default](https://dlang.org/changelog/2.092.0.html#dip25)). DIP1000 enables some idioms that are otherwise forbidden, e.g. pointers to local variables.

Reply:
-dip25 only analyses escaping refs, not scope parameters. But I changed it to 'scope escape analysis...' for clarity.